src/map/battle.c

```diff
@@ -3231,12 +3231,11 @@ static int64 battle_calc_damage(struct block_list *src, struct block_list *bl, s
 		if (!damage) return 0;
 
 		if( (sce = sc->data[SC_LIGHTNINGWALK]) && flag&BF_LONG && rnd()%100 < sce->val1 ) {
-			int dx[8]={0,-1,-1,-1,0,1,1,1};
-			int dy[8]={1,1,0,-1,-1,-1,0,1};
-			uint8 dir = map->calc_dir(bl, src->x, src->y);
-			if( unit->movepos(bl, src->x-dx[dir], src->y-dy[dir], 1, 1) ) {
-				clif->slide(bl,src->x-dx[dir],src->y-dy[dir]);
-				unit->setdir(bl, dir);
+			enum unit_dir dir = map->calc_dir(bl, src->x, src->y);
```

Comment:
using ```enum unit_dir``` is the preferred practice, have a variable type that clearly describe it's expected value is what you should aim for at all times.
saving those 3 bytes in stack is meaningless, such over optimize practices only makes code harder to maintain and shall be only done where it's crucial and needed.
also using unsigned integers is not preferred except for variables meant for bit wise operations.

Reply:
Another note, since it's on stack the compiler would most likely reuse it's space anyway if it's able to, and also ```enums``` variable type can be and might be defined as any integer size by compiler if it thinks it's wise to, not just to int32.

Reply:
^